### PR TITLE
feat: Support non eip155 tx

### PR DESCRIPTION
**Pull request by classicalliu**
Related:
- https://github.com/nervosnetwork/godwoken/pull/777

### CODE REVIEW

File: packages/api-server/src/convert-tx.ts [hunk 1]
```diff
@@ -350,8 +346,13 @@ export async function parseRawTransactionData(
     args_48_52.slice(2) +
     args_data.slice(2);
 
+  let chainId = gwConfig.web3ChainId;
+  // if v = 27 || 28, set chain_id = 0
```
Comment (keroro520) [inline packages/api-server/src/convert-tx.ts:350]:
```suggestion
  // When `v = 27` and `v = 28`, the transaction is considered an non-eip155 transaction.
  //
  // Godwoken has a convention that non-eip155 transactions should be `tx.chain_id == 0`
```

File: packages/api-server/src/convert-tx.ts [hunk 2]
```diff
@@ -132,45 +132,41 @@ export function getSignature(tx: PolyjuiceTransaction): HexString {
...
-    gasPrice: numberToRlpEncode(tx.gasPrice),
-    gasLimit: numberToRlpEncode(tx.gasLimit),
-    value: numberToRlpEncode(tx.value),
-    r: "0x",
-    s: "0x",
-    v: numberToRlpEncode(finalVInt),
-  };
-
-  const encoded = encodePolyjuiceTransaction(rawTx);
+  const encoded: Buffer = rlp.encode(beforeEncode);
```
Comment (keroro520) [inline packages/api-server/src/convert-tx.ts:167]: Creating standalone util functions to make it more readable would be helpful, such as `bool isEip155Transaction`, `bytes rlpEncodeEip155Transaction`, and `bytes rlpEncodeNonEip155Transaction`.

File: packages/api-server/cli/fix-eth-tx-hash.ts [hunk 3]
```diff
@@ -144,8 +144,10 @@ function updateEthTxHash(tx: DBTransaction, chainId: bigint): Hash {
   const r = BigInt("0x" + tx.r.toString("hex"));
   const s = BigInt("0x" + tx.s.toString("hex"));
 
+  // for non eip-155 txs(chain id = 0), v = 27 + last byte of signature(0 / 1)
   // v = chain_id * 2 + 35 + last byte of signature(0 / 1)
```
Comment (keroro520) [inline packages/api-server/cli/fix-eth-tx-hash.ts:148]:
```suggestion
  // for non eip-155 txs(chain id != 0), v = chain_id * 2 + 35 + last byte of signature(0 / 1)
```

**Review by keroro520 — APPROVED** [review]

**Review by RetricSu — CHANGES_REQUESTED** [review]
please edit docs as well https://github.com/nervosnetwork/godwoken-web3/blob/main/docs/compatibility.md#2-signing-transaction-only-support-eip155

File: packages/api-server/src/convert-tx.ts (hunk 1 above)
Comment (classicalliu) [reply packages/api-server/src/convert-tx.ts]: Updated

File: packages/api-server/cli/fix-eth-tx-hash.ts (hunk 3 above)
Comment (classicalliu) [reply packages/api-server/cli/fix-eth-tx-hash.ts]: Updated

**Comment by classicalliu** [comment]
> please edit docs as well https://github.com/nervosnetwork/godwoken-web3/blob/main/docs/compatibility.md#2-signing-transaction-only-support-eip155

Deleted

**Review by RetricSu — APPROVED** [review]

**Pull request merged**